### PR TITLE
epee binary: container can read from empty seq of any type

### PR DESCRIPTION
**Pull request by jeffro256**
Good news: this property holds for current epee binary code in core repo. For real world example of issue brought on if this property is not upheld: https://github.com/monero-rs/monero-epee-bin-serde/issues/49

### CODE REVIEW

**Comment by jeffro256** [comment]
@vtnerd does your new wire reading code safety accept any type marker for empty sequences without error?

**Comment by vtnerd** [comment]
These tests should still work with the new serialization engine. The inner type is only checked when processing that first element.

File: tests/unit_tests/epee_serialization.cpp [hunk 1]
```diff
@@ -54,3 +57,87 @@ TEST(epee_binary, duplicate_key)
...
+  EXPECT_TRUE(epee::serialization::load_t_from_binary(o, epee::span<const std::uint8_t>(data_empty_string)));
+  EXPECT_EQ(0, o.x.size());
+
+  EXPECT_TRUE(epee::serialization::load_t_from_binary(o, epee::span<const std::uint8_t>(data_empty_int64)));
+  EXPECT_EQ(0, o.x.size());
+
+  EXPECT_TRUE(epee::serialization::load_t_from_binary(o, epee::span<const std::uint8_t>(data_empty_object)));
+  EXPECT_EQ(0, o.x.size());
+
+  ObjOfObjs i;
```
Comment (rbrunner7) [inline tests/unit_tests/epee_serialization.cpp:127]: Shouldn't that be `ObjOfInts`?

**Review by rbrunner7 — COMMENTED** [review]
See the single comment

**Review by vtnerd — APPROVED** [review]

File: tests/unit_tests/epee_serialization.cpp (hunk 1 above)
Comment (jeffro256) [reply tests/unit_tests/epee_serialization.cpp]: Yes, good catch ;). Thanks for finding that

**Review by rbrunner7 — APPROVED** [review]

**Pull request merged**